pytorch_lightning/loops/training_batch_loop.py

```diff
@@ -342,8 +342,8 @@ def _process_training_step_output(self, training_step_output: STEP_OUTPUT) -> Op
 
         # handle dict return
         if isinstance(training_step_output, dict):
-            loss = training_step_output.pop("loss", None)
-            hiddens = training_step_output.pop("hiddens", None)
+            loss = training_step_output.get("loss")
+            hiddens = training_step_output.get("hiddens")
```

Comment:
This silently changed in #7871

The change here restores the previous behaviour. Popping here modifies the output argument for the user too in their `training_step_end`